src/App.tsx

```diff
@@ -1,26 +1,114 @@
+  const [selectedTodos, setSelectedTodos] = useState<FilterTypes>(
+    FilterTypes.All,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDeletedTodoHasLoader, setIsDeletedTodoHasLoader] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const areTodosExist = !!todos.length;
+  const notCompletedTodosCount = todos.filter(todo => !todo.completed).length;
+  const isAnyCompletedTodos = notCompletedTodosCount === todos.length;
+
+  const completedIds = todos
+    .filter(todo => todo.completed)
+    .map(todo => todo.id);
+
+  const filteredTodos = filterTodos(todos, selectedTodos);
+  const handleError = (error: string) => {
+    setErrorMessage(error);
+
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        handleError(ErrorMessage.LoadingError);
+      });
+  }, []);
+
+  function handleDeleteTodoCLick(todoId: number) {
+    setIsDeletedTodoHasLoader(true);
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(prevTodos =>
+          prevTodos.filter(currTodo => todoId !== currTodo.id),
+        );
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessage.DeleteTodoError);
+      })
+      .finally(() => setIsDeletedTodoHasLoader(false));
+  }
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div
+      className={classNames('todoapp', {
+        'has-error': errorMessage.length > 0,
+      })}
+    >
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          handleError={handleError}
+          todos={todos}
+          errorMessage={errorMessage}
+          setTempTodo={setTempTodo}
+          setTodos={setTodos}
+        />
+
+        {areTodosExist && (
+          <TodoList
+            todos={filteredTodos}
+            tempTodo={tempTodo}
+            handleDeleteTodoClick={handleDeleteTodoCLick}
+            isDeletedTodoHasLoader={isDeletedTodoHasLoader}
+          />
+        )}
+        {areTodosExist && (
+          <Footer
+            notCompletedTodosCount={notCompletedTodosCount}
+            selectedTodos={selectedTodos}
+            setSelectedTodos={setSelectedTodos}
+            isAnyCompletedTodos={isAnyCompletedTodos}
+            setIsDeletedTodoHasLoader={setIsDeletedTodoHasLoader}
+            completedIds={completedIds}
+            setTodos={setTodos}
+            setErrorMessage={setErrorMessage}
+          />
+        )}
```

Comment:
```suggestion
        {areTodosExist && (
          <TodoList
            todos={filteredTodos}
            tempTodo={tempTodo}
            handleDeleteTodoClick={handleDeleteTodoCLick}
            isDeletedTodoHasLoader={isDeletedTodoHasLoader}
          />
          <Footer
            notCompletedTodosCount={notCompletedTodosCount}
            selectedTodos={selectedTodos}
            setSelectedTodos={setSelectedTodos}
            isAnyCompletedTodos={isAnyCompletedTodos}
            setIsDeletedTodoHasLoader={setIsDeletedTodoHasLoader}
            completedIds={completedIds}
            setTodos={setTodos}
            setErrorMessage={setErrorMessage}
          />
        )}
```
Can we do like this?